model/base_constraint.py

```diff
@@ -173,3 +174,74 @@ def index(self):
     def placements(self):
         """Return the entities where the constraint should be displayed"""
         return []
+
+    def create_slvs_data(self, solvesys: Solver, **kwargs):
+        raise NotImplementedError()
+
+    def py_data(self, solvesys: Solver, **kwargs):
+        return self.create_slvs_data(solvesys, **kwargs)
+
+
+class DimensionalConstraint(GenericConstraint):
+
+    value: Property
+    setting: BoolProperty
+
+    def _set_value(self, val: float):
+        # NOTE: function signature _set_value(self, val: float, force=False)
+        #       will fail when bpy tries to register the value property.
+        #       See `_set_value_force()`
+        if not self.is_reference:
+            self._set_value_force(val)
+    
+    def _set_value_force(self, val: float):
+        self["value"] = val
+
+    def _get_value(self):
+        if self.is_reference:
+            val, _ = self.init_props()
+            return self.to_displayed_value(val)
+        if not self.get("value"):
+            self.assign_init_props()
+        return self["value"]
+
+    def assign_init_props(self, context: Context = None):
+        # self.value, self.setting = self.init_props()
+        _value, _ = self.init_props()
+        self._set_value_force(_value)
+
+    def refresh_constraint(self, context: Context = None):
```

Comment:
Might be great to name this so it reflects that it's only used with reference dimensions...